src/RequestBodyParser.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Request\Body;
+
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Http\Header;
+use Yiisoft\Http\Status;
+use Yiisoft\Request\Body\Parser\JsonParser;
+
+final class RequestBodyParser implements MiddlewareInterface
+{
+    private ResponseFactoryInterface $responseFactory;
+    private ContainerInterface $container;
+
+    private array $parsers = [
+        'application/json' => JsonParser::class,
```

Comment:
What about moving to constants?

Reply:
mime type ? or $parseres ?

Reply:
mime type

Reply:
why not, but it would be better if they were in "yiisoft/http" )

Reply:
I'm not against :)

Reply:
[There](https://www.iana.org/assignments/media-types/media-types.xhtml) is full list of mime types.
Are you sure that we need this?

If you need constants, then you can place them in each corresponding parser.
Linke `JsonParser::PREFERRED_CONTENT_TYPE = 'application/json'`

Reply:
Let's leave it as is for now.